### ISSUE TITLE
Config Files

### ISSUE THREAD

**Issue by YuchenZHANG2** [boschresearch/Hyp-OW#6]
Hi Thang, thanks a lot for the great work and for sharing the code. 

I was wondering if it would be possible to provide the configs for the three splits (OWOD, OWDETR, and Hierarchical). 

I noticed that there is a RUN_HIERARCHICAL_SPLIT.sh script, but I assume this may not be the final config you used, since it seems to be missing "exemplar replay" and the stages for "task 2 finetune, task 3, and task 4". And It would also be really helpful if you could share the configs for the OWOD and OWDETR splits as well.

Thank you very much for your time and help!

**Comment by tldoan** [comment]
Hi Yuchen,

Thanks for your interest! I dug through my laptop, and for training/finetuning it looks like the only parameters that typically change are the related ```--exemplar_replay_selection``` flags and ```--epochs```. You can cross‑check the exact settings in Tables 9 and 10 of the [paper](https://arxiv.org/pdf/2306.14291)


My code was heavily inspired by PROB, so some flags/config structure may match theirs—for [reference](https://github.com/orrzohar/PROB/blob/main/configs/M_OWOD_BENCHMARK.sh)

It’s been a while since I last revisited this, but from memory the snippet would look something like:



```
## training t2
CUDA_LAUNCH_BLOCKING=1  python -u main_open_world.py \
     --output_dir "exps/hypow_hierarchical_t2" --dataset 'HIERARCHICAL'  --PREV_INTRODUCED_CLS 20 --CUR_INTRODUCED_CLS 20 \
     --train_set 't2_train_hierarchical' --test_set 'hierarchical_test' --eval_every 5 --batch_size 3 --lr 1e-4 --num_workers 3  --data_root '/workspace/Hyp-OW/data/OWOD/' \
     --model_type 'hypow'  --obj_temp 1.3  --seed 0 \
     --wandb_name "${WANDB_NAME}_t1" --wandb_project '' --lr_drop 40  --num_queries 100 --logging_freq 40 --use_focal_cls  \
     --save_buffer  --relabel  \
     --epochs 70  --clip_r 1.0 --use_hyperbolic --unknown_weight 0.1  \
     --hyperbolic_c 0.1 --hyperbolic_temp 0.2 --samples_per_category 1 --hyperbolic_coeff 0.05 --checkpoint_period 10 --start_relabelling 0 --emb_per_class 10 \
     --all_background --empty_weight 0.1  --unknown_weight 0.1 --use_max_uperbound  --family_regularizer --family_hyperbolic_coeff 0.02  \
     --pretrain 'checkpoint after 1st task'  --load_buffer
     --exemplar_replay_selection
     
     
## finetuning t2
CUDA_LAUNCH_BLOCKING=1  python -u main_open_world.py \
     --output_dir "exps/hypow_hierarchical_t2_ft" --dataset 'HIERARCHICAL'  --PREV_INTRODUCED_CLS 20 --CUR_INTRODUCED_CLS 20 \
     --train_set 'hierarchical_t2_ft' --test_set 'hierarchical_test' --eval_every 5 --batch_size 3 --lr 1e-4 --num_workers 3  --data_root '/workspace/Hyp-OW/data/OWOD/' \
     --model_type 'hypow'  --obj_temp 1.3  --seed 0 \
     --wandb_name "${WANDB_NAME}_t1" --wandb_project '' --lr_drop 40  --num_queries 100 --logging_freq 40 --use_focal_cls  \
     --save_buffer  --relabel  \
     --epochs 130  --clip_r 1.0 --use_hyperbolic --unknown_weight 0.1  \
     --hyperbolic_c 0.1 --hyperbolic_temp 0.2 --samples_per_category 1 --hyperbolic_coeff 0.05 --checkpoint_period 10 --start_relabelling 0 --emb_per_class 10 \
     --all_background --empty_weight 0.1  --unknown_weight 0.1 --use_max_uperbound  --family_regularizer --family_hyperbolic_coeff 0.02  \
     --pretrain 'checkpoint after 2nd task'  --load_buffer

```  
     

I’ve also attached the set of weights in case you’d like to reproduce the evaluation. Please feel free to reach out if you have any additional questions.

Best,
Thang

**Comment by YuchenZHANG2** [comment]
Hey Thang,

I tried replicating the results for Hierarchical Split (High Regime) Task 1 using the following config:

(The learning rate and batch size match your settings, so I tried to minimize differences in training.)

`/workspace/main_open_world.py --output_dir exps/hypow --dataset HIERARCHICAL --PREV_INTRODUCED_CLS 0 --CUR_INTRODUCED_CLS 20 --train_set t1_train_hierarchical --test_set hierarchical_test --eval_every 20 --batch_size 3 --lr 1e-4 --num_workers 3 --model_type hypow --data_root /workspace/dataset_HUNK/OWOD_Split/data/OWOD/ --wandb_name HYPOW_t1 --lr_drop 40 --num_queries 100 --logging_freq 40 --use_focal_cls --obj_temp 1.3 --save_buffer --relabel --epochs 50 --clip_r 1.0 --use_hyperbolic --unknown_weight 0.1 --hyperbolic_c 0.1 --hyperbolic_temp 0.2 --samples_per_category 1 --hyperbolic_coeff 0.05 --checkpoint_period 10 --start_relabelling 0 --emb_per_class 10 --all_background --empty_weight 0.1 --unknown_weight 0.1 --use_max_uperbound --family_regularizer --family_hyperbolic_coeff 0.02
`

Here are the results I got:
Known mAP: 48.9
Unknown recall: 32.6

<img width="1188" height="956" alt="Image" src="https://github.com/user-attachments/assets/c8650e99-4471-474e-bb52-19586c3f1925" />

Do you see anything I might have set incorrectly?

Best,
Yuchen

**Comment by tldoan** [comment]
Hi Yuchen,


- Sorry for the late reply — it’s been a busy week.
I reviewed the hyperparameters, and at a high level everything looks correct.
- Since this project is from about two years ago, there might be small details I’m forgetting. I can share my older (pre-cleanup) version of the code with you in case it helps spot anything I might have missed — feel free to send me an email.


Best,
Thang
